docs/debugger_v2.md

```diff
@@ -0,0 +1,303 @@
+the completeness is preferred, as in this case, we can disable the circular
+buffer by setting the argument to a negative value (e.g., -1 here).
+
+The debug_mnist_v2 example invokes `enable_dump_debug_info()` by passing
+command-line flags to it. To run our problematic TF2 program again with this
+debugging instrumentation enabled, do:
+
+```sh
+python -m tensorflow.python.debug.examples.v2.debug_mnist_v2 \
+    --dump_dir /tmp/tfdbg2_logdir --dump_tensor_debug_mode FULL_HEALTH
+```
+
+## Starting the Debugger V2 GUI in TensorBoard
+
+Running the program with the debugger instrumentation creates a logdir at
+/tmp/tfdbg2_logdir. We can point start TensorBoard and point at the logdir by:
+
+```sh
+tensorboard --logdir /tmp/tfdbg2_logdir
+```
+
+In the web browser, navigate to TensorBoard’s page at http://localhost:6006. The
+“Debugger V2” plugin should be activated by default, displaying a page that
+looks like the following:
+
+![Debugger V2 full view screenshot](./images/debugger_v2_1_full_view.png)
+
+## Using Debugger V2 GUI to find the root cause of NaNs
+
+The Debugger V2 GUI in TensorBoard is organized into six sections:
+- **Alerts**: This top-left section contains a list of “alert” events detected
+  by the debugger in the debug data generated by the instrumented TensorFlow
+  program. Each alert indicates a certain anomaly that warrants attention. In
+  our case, this section highlights 442 NaN/∞ events with a salient pink-red
+  color. This confirms our suspicion that the model fails to learn because of
+  the presence of NaNs and/or infinities in its internal tensor values. We’ll
+  delve into these alerts shortly.
+- **Python Execution Timeline**: This is the upper half of the top-middle
+  section. It presents the full history of the eager execution of ops and
+  graphs. Each box of the timeline is marked by the initial letter of the op or
+  graph’s name (e.g., “T” for the “TensorSliceDataset” op, “m” for the “model”
+  `tf.function`). We can navigate this timeline by using the navigation buttons
+  and the scrollbar above the timeline.
+- **Graph Execution** : Located at the top-right corner of the GUI, this section
+  will be central to our debugging task. It contains a history of all the
+  floating-dtype tensors computed inside graphs (i.e., compiled by
+  `@tf-function`s).
+- **Graph Structure** (bottom half of the top-middle section), **Source Code**
+  (bottom-left section), and **Stack Trace** (bottom-right section) are
+  initially empty. Their contents will be populated when we interact with the
+  GUI. These three sections will also play important roles in our debugging
+  task.
+
+Having oriented ourselves to the organization of the UI, let’s take the
+following steps to get to the bottom of why the NaNs appeared. First, click the
+**NaN/∞** alert in the Alerts section. This automatically scrolls the list of
+600 graph tensors in the Graph Execution section and focuses on the #67, which
```

Comment:
the image looks like it should be 88, not 67

Reply:
This is where the exact order varies slightly from run to run. Made changes to make the text consistent with the screenshots.